### PR TITLE
TektonInstallerSet: filter resources at once, removes predicates 

### PR DESCRIPTION
**Pull request by sm43**
the order in which resources are applied is imp, for eg. namespace
should be created before configmap and other ns scoped resources.
but we don't need to keep list, hence removing predicates and filtering
resources at first and then applying.

if a new resource kind is added in release yaml, dont need add a predicate
now.

Signed-off-by: Shivam Mukhade <smukhade@redhat.com>

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Run `make test lint` before submitting a PR
- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/operator/blob/master/CONTRIBUTING.md) for more details._

# Release Notes

### CODE REVIEW

**Review by vdemeester — APPROVED** [review]

File: pkg/reconciler/kubernetes/tektoninstallerset/install.go [hunk 1]
```diff
@@ -40,69 +40,97 @@ const (
...
+	// namespace scoped
+	for _, res := range manifest.Resources() {
+		if strings.ToLower(res.GetKind()) == "customresourcedefinition" {
+			installer.crds = append(installer.crds, res)
+			continue
+		} else if res.GetKind() == "Deployment" {
+			installer.deployment = append(installer.deployment, res)
+			continue
+		}
+		if isClusterScoped(res.GetKind()) && strings.ToLower(res.GetKind()) != "clusterrolebinding" {
```
[the remark this answers is not in the document]
Comment (sm43) [reply pkg/reconciler/kubernetes/tektoninstallerset/install.go:76]: manifest client returns error if clusterrole is not on the cluster
so we create clusterrolebinding with namespace scoped resources

[the remark this answers is not in the document]
Comment (sm43) [reply pkg/reconciler/kubernetes/tektoninstallerset/install.go:76]: so that it is created after clusterrole

**Pull request merged**